src/ir/operation.cc

```diff
@@ -23,22 +23,43 @@
 
 namespace raksha::ir {
 
+// Note: I tried to make this only defined in terms of the type parameter T, but
+// the compiler was unhappy with me writing std::function<std::string(const T&)>
+// and I could not figure out how to placate it.
+template<class T, class U>
```

Comment:
`F` is typically used for functors and lambdas?
```suggestion
template<class T, class F>
```

Reply:
OK, I can do that.

Reply:
Done.